crates/holochain_cascade/src/lib.rs

```diff
@@ -623,27 +624,25 @@ where
         options: GetOptions,
     ) -> CascadeResult<Option<EntryDetails>> {
         let authoring = self.am_i_authoring(&entry_hash.clone().into())?;
-        let authority = self.am_i_an_authority(entry_hash.clone().into()).await?;
+        let authority = authoring || self.am_i_an_authority(entry_hash.clone().into()).await?;
         let query: GetEntryDetailsQuery = self.construct_query_with_data_access(entry_hash.clone());
 
-        // We don't need metadata and only need the content
-        // so if we have it locally then we can avoid the network.
-        if let GetStrategy::Content = options.strategy {
-            let results = self.cascading(query.clone()).await?;
-            // We got a result so can short circuit.
-            if results.is_some() {
-                return Ok(results);
-            // We didn't get a result so if we are either authoring
-            // or the authority there's nothing left to do.
-            } else if authoring || authority {
-                return Ok(None);
+        match options.strategy {
+            GetStrategy::Content => {
+                // We don't need metadata and only need the content
+                // so if we have it locally then we can avoid the network.
+                let results = self.cascading(query.clone()).await?;
+                if results.is_some() || authority {
```

Comment:
This means if there is a result, return it. If not and I'm either the authority or am authoring the requested entry, return `None`.
Else, meaning result is `None` and I'm not the authority and not authoring, fetch record from network.

Indeed a bit funny that strategy.